server/src/main/java/org/jetbrains/bsp/bazel/server/sync/BazelProjectMapper.java

```diff
@@ -101,6 +101,10 @@ private Seq<Label> resolveDirectDependencies(TargetInfo target) {
   }
 
   private Set<Language> inferLanguages(TargetInfo target) {
+    if (target.getKind().equals("scala_binary")) {
```

Comment:
is it an edge case for `scala_binary` without sources? the mechanizm below should work for any case with sources

Reply:
cool, maybe it means that we should adjust our mechanism in general - it always should return the language based on the target kind + try to detect more from sources - just a thing to consider later